py/client/pydeephaven/_table_interface.py

```diff
@@ -716,3 +716,41 @@ def where_not_in(self, filter_table: Table, cols: Union[str, List[str]]) -> Unio
         """
         table_op = WhereInTableOp(filter_table=filter_table, cols=to_list(cols), inverted=True)
         return self.table_op_handler(table_op)
+    
+    def slice(self, first_position_inclusive: int, last_position_exclusive: int) -> Union[Table, Query]:
+        """The slice method creates a new table containing rows from the source table, from the first position
+        (inclusive) to the last position (exclusive). 
+
+        - If both first_position and last_position are positive, then both positions are counted from the beginning of 
+          the table.
+        - If both first_position and last_position are negative, then both positions are counted from the end of the 
+          table.
+        - If first_position is negative and last_position is 0, then first_position is counted from the end and the end 
+          is the size of the table.
+        - If first_position is positive and last_position is negative, then first_position is counted from the beginning
+          and last_position is counted from the end.
+        - If first_position is negative and last_position is positive, then first_position is counted from the end and 
+          last_position is counted from the beginning
+
+        Args:
+            first_position_inclusive (int): the first position (inclusive)
+            last_position_exclusive (int): the last position (exclusive)
+
+        Returns:
+            a Table object
+
+        Raises:
+            DHError
+
+        Examples:
+            >>> table.slice(0, 5)    # first 5 rows
+            >>> table.slice(-5, 0)   # last 5 rows
+            >>> table.slice(2, 6)    # rows from index 2 to 5
+            >>> table.slice(6, 2)    # ERROR: cannot slice start after end
+            >>> table.slice(-6, -2)  # rows from 6th last to 2nd last (exclusive)
+            >>> table.slice(-2, -6)  # ERROR: cannot slice start after end
+            >>> table.slice(2, -3)   # all rows except the first 2 and the last 3
+            >>> table.slice(-6, 8)   # rows from 6th last to index 8 (exclusive)
```

Comment:
The docs differ significantly from the other two interfaces.  It isn't obvious that they should differ.  Combine the best parts of both?

@jmao-denver agree?

Reply:
Yes. I'd vote for reverting the names of the parameters and combining the best parts of both. Also The server API has a `slice_pct` variant. Should we add that to the clients as well?

Reply:
Let's defer `slice_pct` to a separate ticket, it also isn't exposed via grpc, etc.